### PR TITLE
feat(ProfileShowcase): Implement custom showcase position when searcher text is filtering items

### LINKED ISSUE #13682: [Profile Showcase] Custom positioning when reordering a filtered `In showcase` list
### Description

[Design](https://www.figma.com/file/ibJOTPlNtIxESwS96vJb06/👤-Profile-%7C-Desktop?type=design&node-id=4592-433588&mode=design&t=8Y8EirJBCMdWp4vW-0)

The user is free to drag items into the showcase and re-order whilst in the search filtered state.   We should follow Sheets / Excel logic for defining the relative position of moved items once the search filter is removed.   In essence this logic is as follows:

Moving an item down the list while filter applied When the filter is removed the item is left directly under under the item it was moved below while in the filtered state (we can only infer which of the visible list items they wish to move the list item below). 
Moving an item up the list while filter applied When the filter is removed the item is left directly above the item it is moved above while in the filtered state (we can only infer which of the visible list items they wish to move the list item above).

Note for development: Re-enable the DND

### LINKED ISSUE #14040: Profile showcase management: Search shouldn't affect the preview
# Bug Report

## Description

The filtering done via search bar should be local thing to particular tabs, not affecting external world, preview in particular. Ideally this logic should be moved to the view from `ProfileShowcaseDirtyState.qml`.

Note: Currently `ProfileShowcaseDirtyState.qml` in `changePosition` doesn't take this search sfpm when mapping to source what can cause some undesired behaviors.

[Kazam_screencast_00098.webm](https://github.com/status-im/status-desktop/assets/20650004/f648dc77-785d-4acb-9a65-e6299102102c)

## Steps to reproduce
1. Go to Settings -> Profile -> Accounts
2. Make window wider to activate preview
3. Use the search bar

#### Expected behavior
Search doesn't affect preview.

#### Actual behavior
Preview is also filtered.

### PR DESCRIPTION
**Pull request by alexjba**
### What does the PR do

closes #13682
closes #14040

Changes:
1. Move the searcher SFPM in the ProfileShowcasePanel along with the MovableModel
2. Compute the indexes involved in the position change for all proxy layers
3. Fully implement changePosition in `VisibilityAndPositionDirtyStateModel.qml`
4. Fix storybook pages

### Affected areas

Profile showcase

### Screenshot of functionality (including design for comparison)

https://github.com/status-im/status-desktop/assets/47811206/364bb73e-3267-4f2d-aac5-7016c12d312d

### CODE REVIEW

File: ui/app/AppLayouts/Profile/panels/ProfileShowcaseAccountsPanel.qml [hunk 1]
```diff
@@ -30,4 +30,13 @@ ProfileShowcasePanel {
         icon.color: model && model.colorId ? Utils.getColorForId(model.colorId) : Theme.palette.primaryColor3
         highlighted: model ? model.address === root.currentWallet : false
     }
+    filter: FastExpressionFilter {
+        expression: {
+            root.searcherText
+            return (address.toLowerCase().includes(root.searcherText.toLowerCase()) ||
```
Comment (caybro) [inline ui/app/AppLayouts/Profile/panels/ProfileShowcaseAccountsPanel.qml:36]: You could extract the `searcherText.toLowerCase()` expression somewhere in order not to re-evaluate it multiple times (same for other files)

**Review by caybro — APPROVED** [review]
LGTM

**Review by micieslak — COMMENTED** [review]
I really like the approach, it organizes things nicely. Only some small questions here.

And as a future ideas:
- bulk set(...) method for WritableModel - it would prevent from many dataChange signals and other reevaluations later in the flow
- custom proxy just for DnD, simpler than MovableModel, converting dataChanged from source into rowsMoved signals

**Comment by alexjba** [comment]
@micieslak Good point!

I've updated this PR and created two additional tasks:
https://github.com/status-im/status-desktop/issues/14727
https://github.com/status-im/status-desktop/issues/14728

Feel free to update them.

File: ui/StatusQ/src/StatusQ/Core/Utils/ModelUtils.qml [hunk 2]
```diff
@@ -44,6 +44,7 @@ QtObject {
                 else if (entry !== undefined)
                     arrayItem[role] = entry
             })
+            arrayItem.index = i
```
[the remark this answers is not in the document]
Comment (alexjba) [reply ui/StatusQ/src/StatusQ/Core/Utils/ModelUtils.qml:47]: yes, removing it now

[the remark this answers is not in the document]
Comment (alexjba) [reply ui/StatusQ/src/StatusQ/Core/Utils/ModelUtils.qml]: done!

**Review by micieslak — APPROVED** [review]

**Pull request merged**